deepmd/tf/descriptor/se_atten.py

```diff
@@ -718,6 +718,18 @@ def _pass_filter(
                 tf.shape(inputs_i)[0],
                 self.nei_type_vec,  # extra input for atten
             )
+            #  (nsamples * natoms * nnei, 1)
+            nei_exclude_mask = tf.slice(
+                tf.reshape(tf.cast(mask, self.filter_precision), [-1, 4]),
+                [0, 0],
+                [-1, 1],
+            )
+            #  (nsamples * natoms, 1,  nnei)
+            self.nmask *= tf.reshape(
+                nei_exclude_mask,
+                [-1, 1, self.sel_all_a[0]],
+            )
+            self.negative_mask = -(2 << 32) * (1.0 - self.nmask)
```

Comment:
To my understanding, `nmask` and `negative_mask` are only used when `smooth` is False, right?

Reply:
Yes you're correct. I've placed them in False condition.